core/GlobalState.cc

```diff
@@ -1492,7 +1493,9 @@ bool GlobalState::wasModified() const {
 }
 
 bool GlobalState::wasTypecheckingCanceled() const {
-    return lspEpochInvalidator->load() != currentlyProcessingLSPEpoch->load();
+    // Can't cancel typechecking if we have a preemption task to run -- we have to make it past resolver at least before
+    // we can cancel.
+    return lspEpochInvalidator->load() != currentlyProcessingLSPEpoch->load() && !atomic_load(preemptTask.get());
```

Comment:
1. can we make a `nullptr` comparison explicit
2. I didn't understand how does checking for `preemptTask` being not null relate to comment of 'we have to make it past resolver at least before we can cancel'. Are there some backed-in assumptions here?

Reply:
1. Sure!
2. I see the confusion here; I can update the comment to remove the resolver bit. "Can't cancel typechecking if we have a preemption task to run" captures the behavior completely. I wrote the resolver comment because we don't run preemption tasks until after resolver, and it would cause bugs if we canceled before resolver and _then_ ran the preeemption task.